docs/tutorials/Setting-up-Raspberry-Pi-Zero-W.md

```diff
@@ -0,0 +1,153 @@
+cd OpenBLAS
+make
+sudo make install
+```
+
+The whole build process will take about 4-5 hours. The above steps will install a working copy of OpenBLAS to /opt/OpenBLAS
+
+### Numpy
+Numpy is a numerical processing package required by OpenCV and our tutorial. Here we will install a copy of Numpy package which uses the OpenBlas we installed in the last step:
+
+```shell
+git clone https://github.com/numpy/numpy.git numpy
+cd numpy
+```
+
+create a new file called sit.cfg under numpy directory and type in following contents:
+
+```shell
+[default]
+include_dirs = /opt/OpenBLAS/include
+library_dirs = /opt/OpenBLAS/lib
+ 
+[openblas]
+openblas_libs = openblas
+library_dirs = /opt/OpenBLAS/lib
+ 
+[lapack]
+lapack_libs = openblas
+library_dirs = /opt/OpenBLAS/lib
+```
+
+now still under the same directory, runt he following commands:
+
+```shell
+python3 setup.py build --fcompiler=gnu95
+python3 setup.py install
+```
+
+### curl
+`curl` is a command line tool used to transfer data via URL. When files are required to be downloaded from a URL, the instructions assume you have `curl` available to perform the download. To install `curl`, type the following:
+```shell
+sudo apt-get install -y curl
+```
+
+Remember to run `python3 xxxx.py` when you run any python scripts from ELL tutorial.
+
+### OpenCV
+[OpenCV](http://opencv.org/) is a computer vision library that enables us to read images from a camera, resize them, and prepare them for processing by ELL. We cannot install it easily through Conda like we do for Raspberry Pi 3 since we don't have a conda tool here, so we have to install it manually again. Now take a deep breath, this will take you lots of time.
+
+Before we start building OpenCV source, we need to install a few tools. Install them by following the steps below:
+
+```shell
+sudo apt-get install build-essential pkg-config
+sudo apt-get install libjpeg-dev libtiff5-dev libjasper-dev libpng12-dev
+sudo apt-get install libavcodec-dev libavformat-dev libswscale-dev libv4l-dev
+sudo apt-get install libxvidcore-dev libx264-dev
+sudo apt-get install libgtk2.0-dev libgtk-3-dev
+sudo apt-get install libatlas-base-dev gfortran
+```
+
+OK, now we'll download the OpenCV source and get into our OpenCV installation business.
+
+```shell
+cd ~
+wget -O opencv.zip https://github.com/Itseez/opencv/archive/3.3.0.zip
+unzip opencv.zip
+wget -O opencv_contrib.zip https://github.com/Itseez/opencv_contrib/archive/3.3.0.zip
+unzip opencv_contrib.zip
+cd ~/opencv-3.3.0/
+mkdir build
+cd build
+cmake -D CMAKE_BUILD_TYPE=RELEASE \
+    -D CMAKE_INSTALL_PREFIX=/usr/local \
+    -D INSTALL_PYTHON_EXAMPLES=ON \
+    -D OPENCV_EXTRA_MODULES_PATH=~/opencv_contrib-3.3.0/modules \
+    -D BUILD_EXAMPLES=ON ..
+```
+
+Check the output of the CMake and make sure the "Python 3" section includes correct path to your environment. If everything looks fine, type in the following command in the OpenCV directory:
+
+```shell
+make
```

Comment:
compiler keeps running out of memory or getting internal compiler error. I'm trying the following steps:

sudo nano /etc/dphys-swapfile

CONF_SWAPSIZE=1024

sudo /etc/init.d/dphys-swapfile stop
sudo /etc/init.d/dphys-swapfile start

Reply:
For some reason, I didn't see any memory issue (or maybe I missed it). Yeah it's nice to have this set up before hand. I added a new section to instruct user setting up swap space before all package installation.
Thanks!